rules/terraformrules/terraform_naming_convention.go

```diff
@@ -37,6 +46,19 @@ type NameValidator struct {
 	Regexp        *regexp.Regexp
 }
 
+// RuleTarget contains information of a construct (e.g. data block, local value) whose name is checked by the rule
```

Comment:
Please make sure to match upstream names. What you've labeled "construct" is official a "block" or "block type":

https://www.terraform.io/docs/configuration/syntax.html#blocks